Add preference to use system native file dialogs

williamchange opened the pull request:
- Resolves https://github.com/RodZill4/material-maker/issues/643

https://github.com/user-attachments/assets/742f3702-ea79-4e15-9c51-156def73cab5

https://github.com/user-attachments/assets/129d2fcb-3d8e-47ba-b0f5-88294585d687

williamchange commented:
On Linux (Fedora 42) it's a bit buggy, probably just enabling it isn't enough

https://github.com/user-attachments/assets/d46d6598-1601-46ea-a771-4d94d43a6de4

Calinou commented on material_maker/globals.gd:
```diff
@@ -30,6 +30,7 @@ const DEFAULT_CONFIG : Dictionary = {
 	auto_size_comment = true,
 	graph_line_curvature = 0.5,
 	graph_line_style = 1,
+	ui_use_native_file_dialogs = false,
```
I suggest enabling this by default, as it's what most people expect for GUI applications these days.

williamchange commented:
Addressed review comment

> What is the current status of this?

I'm mostly waiting for @RodZill4 to merge the following since it's pending in the milestone and it will result in a conflict if this got merged first (most preferences-related PRs will, so I am marking them as drafts for now)
- https://github.com/RodZill4/material-maker/pull/969

williamchange commented:
This could probably be a blocker (mac)
- https://github.com/godotengine/godot/issues/113685
